### DISCUSSION TITLE
Create Message is been made as internal, is there any workaround

### DISCUSSION THREAD

**Discussion by navichandan** [CoreWCF/CoreWCF#672] [Q&A] [answered]
Hi,

Our service returns a message on any failures, and based on the fault code and reason the client takes certain actions. 
I see the below method been made as internal and cant use this further. 

        internal static Message CreateMessage(MessageVersion version, FaultCode faultCode, string reason, string action)

Later tried to use below instead above 
        public static Message CreateMessage(MessageVersion version, MessageFault fault, string action)
but in above case we have CreateFault method as internal again.
        internal static MessageFault CreateFault(FaultCode code, string reason)
     
Is there any workaround for this, client is not willing to change any code.

**Comment by mconnew** [comment, marked as answer]
You can work around this, you are really close in your approach:
```c#
var fault = MessageFault.CreateFault(faultCode, new FaultReason(reason), null, null, "", "");
var message = Message.CreateMessage(version, fault, action);
```

We can make these api's public. This code was copied from the WCF client and they weren't public there. They are only internal because nobody has asked for them yet. I'm going to create an issue from this discussion to track making these api's public.

**Reply by navichandan** [reply]
Thank you so much on taking it forward. 

Yes, did the same and need to test further from client side.
